test/integration_test/migration_features_test.go

```diff
@@ -0,0 +1,336 @@
+	require.NoError(t, err, fmt.Sprintf("Error activating app in namespace %s", appData.Ns))
+
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "Error setting dest kubeconfig")
+
+	logrus.Infof("Waiting for application pods to come up")
+	time.Sleep(5 * time.Minute)
+	logrus.Infof("Applications pods should be up by now")
+
+	// validate that CR is present
+	validated, err = validateCR(appName, appData.Ns, destClusterConfigPath)
+	require.NoError(t, err, fmt.Sprintf("Error validating CR for app %s in namespace %s", appName, appData.Ns))
+	require.True(t, validated, fmt.Sprintf("CR for app %s in namespace %s should have been present", appName, appData.Ns))
+
+	podsMigrated, err := core.Instance().GetPods(appData.Ns, nil)
+	require.NoError(t, err, "Error getting migrated pods")
+
+	podsMigratedLen := len(podsMigrated.Items)
+	require.Equal(t, podsCreatedLen, podsMigratedLen, "Pods migration failed as len of pods found on source doesnt match with pods found on destination")
+
+	logrus.Infof("Delete Destination and Source Ns")
+	err = core.Instance().DeleteNamespace(appData.Ns)
+	if err != nil {
+		logrus.Errorf("Error deleting namespace %s: %v\n", appData.Ns, err)
+	}
+	err = setSourceKubeConfig()
+	require.NoError(t, err, "Error setting source kubeconfig")
+
+	err = asyncdr.DeleteAndWaitForMigrationDeletion(mig.Name, mig.Namespace)
+	require.NoError(t, err, "Error deleting migration")
+
+	asyncdr.DeleteCRAndUninstallCRD(appData.OperatorName, appPath, appData.Ns)
+	err = core.Instance().DeleteNamespace(appData.Ns)
+	if err != nil {
+		logrus.Errorf("Error deleting namespace %s: %v\n", appData.Ns, err)
+	}
+	logrus.Infof("Test %s ended", t.Name())
+}
+
+func updateAppReg(appName string, suspendOptions storkapi.SuspendOptions, stashStrategy storkapi.StashStrategy) error {
+	operatorCRMap := getSupportedOperatorCRMapping()
+	if _, ok := operatorCRMap[appName]; !ok {
+		return fmt.Errorf("app %s is not currently supported in test framework", appName)
+	}
+	appResources := operatorCRMap[appName]
+
+	for _, appResource := range appResources {
+		appregName := strings.ToLower(appResource.Kind)
+		reg, err := storkops.Instance().GetApplicationRegistration(appregName)
+		if err != nil {
+			return fmt.Errorf("error getting application registration %s: %w", appregName, err)
+		}
+		for i, resource := range reg.Resources {
+			resource.SuspendOptions = suspendOptions
+			resource.StashStrategy = stashStrategy
+			reg.Resources[i] = resource
+		}
+		_, err = storkops.Instance().UpdateApplicationRegistration(reg)
+		if err != nil {
+			return fmt.Errorf("error setting stash strategy in application registration %s, %w", appregName, err)
+		}
+	}
+	return nil
+}
+
+func validateCR(appName string, namespace string, kubeConfig string) (bool, error) {
+	validated := true
+	operatorCRMap := getSupportedOperatorCRMapping()
+	if _, ok := operatorCRMap[appName]; !ok {
+		return validated, fmt.Errorf("app %s is not currently supported in test framework", appName)
```

Comment:
nit.. We are returning err and handling in caller. But returning validated = true is a bit misleading. We should return false in all error scenarios.

Reply:
Addressed.